### PR TITLE
Feature/pokemon brunch

### PR DESCRIPTION
**Pull request by RomiKMoveo**
### Description
This pull request addresses the task described in the exercise by implementing a minisite that displays the first 100 Pokemons in card form. Each card includes the Pokemon's image and name, and is clickable to show more information about the selected Pokemon. Additionally, a search system by name and a filter option by type have been implemented as bonus features.

### Changes Made
Created a new Angular project.
Integrated SCSS for styling and TypeScript for functionality.
Utilized the PokeAPI to fetch data for the first 100 Pokemons.
Designed card components to display Pokemon image and name.
Implemented click functionality on cards to display detailed information about the selected Pokemon.
Added a search system allowing users to search for Pokemons by name.
Implemented filtering functionality to filter Pokemons by type.

### Related Issues

I have:
- [ ] deleted all irrelevant comments, logs and debuggers.
- [ ] assure that I don't print full error stacks, full body response or headers from the server in the console.
- [ ] deleted all sensitive information from the code because it's being imported from environment variables.
- [ ] encrypted keys/passwords that are sent through the internet.
- [ ] sanitized all user inputs to prevent attacks.
- [ ] made sure my code is generic and reusable for future things or reused existing resources.
- [ ] made sure that my code is readable.
- [ ] checked that the functions and resource names are self-explanatory.
- [ ] tested the code locally and it works as expected.
- [ ] added validations and error handling in relevant cases (external requests, integrations, etc).

### CODE REVIEW

File: pokemon-minisite/src/app/app.component.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  selector: 'app-root',
+  standalone: true,
+  imports: [RouterOutlet, HeaderComponent, PokemonListComponent],
+  templateUrl: './app.component.html',
+  styleUrl: './app.component.scss'
+})
+export class AppComponent implements OnInit{
+  title = 'pokemon-minisite';
+
+  ngOnInit(): void {
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/app.component.ts]: delete oninit function and import if not in use

File: pokemon-minisite/src/app/components/header/header.component.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+.header {
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/header/header.component.scss:1]: please prettier this file.
you can use automatic prettier tool in vs code

File: pokemon-minisite/src/app/components/header/header.component.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+@Component({
+  selector: 'app-header',
+  standalone: true,
+  imports: [],
+  templateUrl: './header.component.html',
+  styleUrl: './header.component.scss'
+})
+export class HeaderComponent {
+  constructor() {}
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/header/header.component.ts]: delete empty functions

File: pokemon-minisite/src/app/components/pokemon-list/filter-pokemon.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { PokemonInterface } from '../../interfaces/pokemon.interface';
+
+@Pipe({
+  name: 'filterPokemon',
+  standalone: true
+})
+export class FilterPokemonPipe implements PipeTransform {
+  transform(pokemons: PokemonInterface[], selectedType: string, searchInput: string): any[] {
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/pokemon-list/filter-pokemon.pipe.ts]: please give more specific name to the function

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-detail/pokemon-detail.component.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public selectedPokemon: PokemonInterface,
+    private pokemonService: PokemonService) 
+  {
+    this.pokemonToPresent = selectedPokemon; 
+  }
+
+    
+    ngOnInit(): void {
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/pokemon-list/pokemon-detail/pokemon-detail.component.ts]: delete empty function

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.scss [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    .card__info {
+       
+        flex-direction: column;
+        color: $rock;
+        text-transform: capitalize;
+        text-align: center;
+
+        h2 {
+            border-radius: 40px;
+            font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.scss]: please use variable font

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+import { PokemonDetailComponent } from './pokemon-detail/pokemon-detail.component';
+import { FilterPokemonPipe } from './filter-pokemon.pipe';
+
+
+
+
+@Component({
+  selector: 'app-pokemon-list',
+  standalone: true,
+  imports: [
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts:17]: i dont think you need it .

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+import { MatDialog } from '@angular/material/dialog';
+import { PokemonDetailComponent } from './pokemon-detail/pokemon-detail.component';
+import { FilterPokemonPipe } from './filter-pokemon.pipe';
+
+
+
+
+@Component({
+  selector: 'app-pokemon-list',
+  standalone: true,
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts:16]: also this

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    .subscribe((pokemons: PokemonInterface[]) => {
+      this.pokemons = pokemons;
+
+      this.getAllTypes();  
+    });
+    
+    }
+
+    openPopup(pokemon: PokemonInterface): void {
+      const dialogRef = this.dialog.open(PokemonDetailComponent, {
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts:50]: you dont need const dialogRef

File: pokemon-minisite/src/app/services/pokemon.service.ts [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+export class PokemonService {
+  private readonly apiUrl = `https://pokeapi.co/api/v2/pokemon/`;
+  private readonly imageUrl = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/';
+
+  constructor(private http: HttpClient) {}
+
+  getPokemonList(limit: number = 100): Observable<PokemonInterface[]> {
+    return this.http.get<any>(this.apiUrl + '?offset=0&limit=' + limit).pipe(
+      map((response: any) =>
+        response.results.map((pokemon: any) => ({
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/services/pokemon.service.ts:21]: please create type for this pokemon

File: pokemon-minisite/src/app/services/pokemon.service.ts [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+          id: this.extractPokemonIdFromUrl(pokemon.url),
+          image: `${this.imageUrl}${this.extractPokemonIdFromUrl(pokemon.url)}.png`
+        }))
+      ),
+      mergeMap((pokemons: PokemonInterface[]) => {
+        const requests: Observable<any>[] = pokemons.map((pokemon: { url: string }) =>
+          this.http.get<PokemonInterface>(pokemon.url).pipe(
+            map((detail: PokemonInterface) => ({
+              types: detail.types.map((type: any) => type.type.name),
+              height: detail.height, // Include height property
```
Comment (sivanbasa93) [inline pokemon-minisite/src/app/services/pokemon.service.ts]: please delete comment

File: pokemon-minisite/src/app/variables.scss
Comment (sivanbasa93) [inline pokemon-minisite/src/app/variables.scss:1]: create folder "style" and put variables file there

File: pokemon-minisite/src/app/components/pokemon-list/filter-pokemon.pipe.ts
Comment (sivanbasa93) [inline pokemon-minisite/src/app/components/pokemon-list/filter-pokemon.pipe.ts:1]: create folder pipes and put this pipe there

Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/pokemon-list/filter-pokemon.pipe.ts:1]: done

File: pokemon-minisite/src/app/variables.scss
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/variables.scss:1]: done

File: pokemon-minisite/src/app/services/pokemon.service.ts (hunk 11 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/services/pokemon.service.ts]: done

File: pokemon-minisite/src/app/app.component.ts (hunk 1 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/app.component.ts]: done

File: pokemon-minisite/src/app/components/header/header.component.scss (hunk 2 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/header/header.component.scss:1]: done

File: pokemon-minisite/src/app/components/header/header.component.ts (hunk 3 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/header/header.component.ts]: done

File: pokemon-minisite/src/app/components/pokemon-list/filter-pokemon.pipe.ts (hunk 4 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/pokemon-list/filter-pokemon.pipe.ts]: done

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts (hunk 7 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts:17]: done

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts (hunk 8 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.ts:16]: done

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-detail/pokemon-detail.component.ts (hunk 5 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/pokemon-list/pokemon-detail/pokemon-detail.component.ts]: done

File: pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.scss (hunk 6 above)
Comment (RomiKMoveo) [reply pokemon-minisite/src/app/components/pokemon-list/pokemon-list.component.scss]: done

**Pull request merged**